zone/embparser.cpp

```diff
@@ -1555,6 +1556,17 @@ void PerlembParser::ExportEventVariables(
 			break;
 		}
 
+		case EVENT_SERVERFIRST_LEVEL: {
+		if (mob) {
+			ExportVar(package_name.c_str(), "name", mob->GetName());
+			ExportVar(package_name.c_str(), "race", GetRaceIDName(mob->GetRace()));
+			ExportVar(package_name.c_str(), "class", GetClassIDName(mob->GetClass()));
+			ExportVar(package_name.c_str(), "ulevel", mob->GetLevel());
```

Comment:
Unnecessary, as $name/$class/$race/$ulevel/$userid are always exported in client scripts.

If not, can quest::getracename($client->GetRace()), quest::getclassname($client->GetClass()), $client->GetLevel(), $client->GetCleanName() or $client->GetID().